src/Mono.Android/Android.Runtime/AndroidRuntime.cs

```diff
@@ -234,7 +234,7 @@ protected override IEnumerable<string> GetSimpleReferences (Type type)
 			foreach (var simpleRef in base.GetSimpleReferences (type)) {
 				yield return simpleRef;
 			}
-			var j = JNIEnv.monodroid_typemap_managed_to_java (type.FullName + ", " + type.Assembly.GetName ().Name);
+			var j = JNIEnv.monodroid_typemap_managed_to_java (type.Module.ModuleVersionId.ToByteArray (), type.MetadataToken);
```

Comment:
What is funny is I just found out how bad `Assembly.GetName().Name` is: https://github.com/jonathanpeppers/Benchmarks/tree/assembly.getname

Taking that out is definitely going to help, and I won't need to send this in 👍 : https://github.com/xamarin/xamarin-android/commit/cb40999178a12fc2eaf81d914c6cd7833d75047d

Reply:
Another thought, is it worth keeping a cache/map of each`ModuleVersionId`, so `ModuleVersionId.ToByteArray()` would only be called once per assembly (module)?

Otherwise this will allocate a `byte[]` for a `Guid` on every call.

Reply:
Yes, the cache is an idea I want to investigate at some point, but not in this PR as it's not as easy as keeping everything we collect - it might use too much RAM. We need an eviction algorithm.

Reply:
The best option would be if there was some way to use [ArrayPool](https://docs.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1) for the `byte[]` here: 

https://github.com/mono/mono/blob/f4567e83dde9393fe19d4cb7e315c21899c67a3f/mcs/class/referencesource/mscorlib/system/guid.cs#L892-L914

Reply:
Or if `Guid` had a `ToArray(byte[])` - then we could just reuse a single array per thread

Reply:
There is a [`Guid.TryWriteBytes(Span<byte>)` method](https://docs.microsoft.com/en-us/dotnet/api/system.guid.trywritebytes?view=netcore-3.1), in .NET Standard 2.1, and *is* present in our BCL.

Reply:
That's worth investigating, thanks

Reply:
The code now uses `Guid.TryWriteBytes`